source/fab/util.py

```diff
@@ -127,18 +147,67 @@ class CompiledFile(object):
     A Fortran or C file which has been compiled.
 
     """
-    def __init__(self, analysed_file, output_fpath: Path):
+    def __init__(self, input_fpath, output_fpath,
+                 source_hash=None, flags_hash=None, module_deps_hashes: Dict[str, int] = None):
         """
-        :param analysed_file:
+        :param input_fpath:
             The file that was compiled.
         :param output_fpath:
-            The object file.
+            The object file that was created.
+
+        """
+        # todo: Should just be the input_fpath, not the whole analysed file
+        self.input_fpath = Path(input_fpath)
+        self.output_fpath = Path(output_fpath)
+
+        self.source_hash = source_hash or 0
+        self.flags_hash = flags_hash or 0
+        self.module_deps_hashes = module_deps_hashes or {}
+
+    #
+    # persistence
+    #
+    @classmethod
+    def field_names(cls):
```

Comment:
Names for which fields? Presumably the persistence database. It would be good to properly encapsulate the persistance at some point.

Reply:
Agreed - both for this and the analysis data. Disappointed [dbm](https://docs.python.org/3/library/dbm.html) wasn't up to the job!

Reply:
In the meantime can you rename this function just to make it clear what is being named.